app/javascript/components/supplemental_application/SupplementalApplicationPage.js

```diff
@@ -167,6 +178,73 @@ class SupplementalApplicationPage extends React.Component {
     }
   }
 
+  handleCloseRentalAssistancePanel = () => {
+    this.setState({ showAddRentalAssistanceBtn: true, showNewRentalAssistancePanel: false })
+  }
+
+  handleSaveNewRentalAssistance = (rentalAssistance) => {
+    // TODO: Add a real call to Salesforce to save the new
+    // rental assistance record and get back the ID of the
+    // record. Here, we mock the real Salesforce save by
+    // adding a fake ID to the rental assistance values.
+    const savedRentalAssistance = ((rentalAssistance) => {
+      rentalAssistance.id = sampleSize('ABCDEFGabcdefg0123456789', 6).join('')
+      return rentalAssistance
+    })(rentalAssistance)
+
+    this.setState(prev => {
+      return {
+        rentalAssistances: [...prev.rentalAssistances, savedRentalAssistance],
+        showNewRentalAssistancePanel: false,
+        showAddRentalAssistanceBtn: true
+      }
+    })
+  }
+
+  handleUpdateRentalAssistance = (rentalAssistance) => {
+    // Set other assistance name to null of Other is not selected (Reset to blank).
```

Comment:
thanks for adding this update @fddayan! the following two lines are pretty clear, so i think we can do away with this comment